api/v1alpha1/backendtrafficpolicy_types.go

```diff
@@ -74,11 +73,26 @@ type BackendTrafficPolicySpec struct {
 	//
 	// +optional
 	ResponseOverride []*ResponseOverride `json:"responseOverride,omitempty"`
+	// UpgradeConfigs defines the configuration for protocol upgrades.
+	// If not specified, the default upgrade configuration(websocket) will be used.
+	//
+	// +optional
+	UpgradeConfigs []*BackendUpgradeConfig `json:"upgradeConfigs,omitempty"`
 }
 
-// +kubebuilder:object:root=true
+type BackendUpgradeConfig struct {
+	// UpgradeType is the case-insensitive name of protocol upgrade.
+	//
+	// +kubebuilder:validation:Required
+	UpgradeType string `json:"upgradeType"`
```

Comment:
prefer `type`, since we've used `upgrade` earlier in the parent name

Reply:
we should also consider enum here

Reply:
prefer to `backendProtocol` or `protocol` than type here.
Keep it as string will give us free to support new/unspported protocol in the future.

Reply:
does envoy proxy handle arbitrary string well ?

Reply:
the problem is that envoy didn't list protocols it supported in doc.